File: 002.직사각형 네개의 합집합의 면적 구하기/area.kt
import kotlin.system.exitProcess
//import java.util.Arrays

fun main() {
  var inputArr = Array(4) {Array(4) {0} }
  //println(inputArr.javaClass.kotlin.simpleName)
  
  // Get Input
  for(i: Int in 0..3) {
    var input = readLine() ?: exitProcess(0)
    //println(input.javaClass.kotlin.simpleName)
    
    var splitArr = input.split(" ").map { it.toInt() }
    //println(splitArr.javaClass.kotlin.simpleName)
    
    inputArr[i] = splitArr.toTypedArray()
    //println(splitArr.toTypedArray().javaClass.kotlin.simpleName)
  }
  
  /* Print Input
  for (item in inputArr) {
    println(Arrays.toString(item))
  }
   */
  var result: Int = 0
  for (x in 1..100) {
    for (y in 1..100) {
      for (z in 0..3) {
        if (x >= inputArr[z][0] && x < inputArr[z][2] &&
              y >= inputArr[z][1] && y < inputArr[z][3]) {
          result += 1
          break
        }
      }
    }
  }
  
  println(result)
}

File: 002.직사각형 네개의 합집합의 면적 구하기/area.py
inputList = []

for n in range(4):
  inputArr = input()
  tmpList = list(map(int, inputArr.strip().split(' ')))
  inputList.append(tmpList)
print(inputList)

result = 0

for x in range(101):
  if (x == 0):
    continue
  for y in range(101):
    if (y == 0):
      continue
    for z in range(4):
      if (x >= inputList[z][0] and x < inputList[z][2]):
        if (y >= inputList[z][1] and y < inputList[z][3]):
          result += 1
          break 
print(result)